operator/controllers/execution/scans/hook_reconciler.go

```diff
@@ -62,250 +60,232 @@ func (r *ScanReconciler) setHookStatus(scan *executionv1.Scan) error {
-	// filter all ReadOnlyHooks in the scamCompletionHooks list
-	for _, hook := range scanCompletionHooks.Items {
-		if hook.Spec.Type == executionv1.ReadOnly {
-			readOnlyHooks = append(readOnlyHooks, hook)
+	var err error
+	switch nonCompletedHook.State {
+	case executionv1.Pending:
+		var hook executionv1.ScanCompletionHook
+		err = r.Get(ctx, types.NamespacedName{Name: nonCompletedHook.HookName, Namespace: scan.Namespace}, &hook)
+		if err != nil {
+			r.Log.Error(err, "Failed to get ReadAndWrite Hook for HookStatus")
+			break
 		}
-	}
 
-	r.Log.Info("Found ReadOnlyHooks", "ReadOnlyHooks", len(readOnlyHooks))
-
-	// If the readOnlyHooks list is empty, nothing more to do
-	if len(readOnlyHooks) == 0 {
-		r.Log.Info("Marked scan as done as without running ReadOnly hooks as non were configured", "ScanName", scan.Name)
-		scan.Status.State = "Done"
-		var now metav1.Time = metav1.Now()
-		scan.Status.FinishedAt = &now
-		if err := r.Status().Update(ctx, scan); err != nil {
-			r.Log.Error(err, "Unable to update Scan status")
-			return err
+		var jobs *batch.JobList
+		jobs, err = r.getJobsForScan(scan, client.MatchingLabels{
+			"securecodebox.io/job-type":  jobType,
+			"securecodebox.io/hook-name": nonCompletedHook.HookName,
+		})
+		if err != nil {
+			break
 		}
-		return nil
-	}
-
-	// Get all read-only-hooks for scan to later check that they weren't already created
-	jobs, err := r.getJobsForScan(scan, client.MatchingLabels{
-		"securecodebox.io/job-type": "read-only-hook",
-	})
-	if err != nil {
-		return err
-	}
-
-	for _, hook := range readOnlyHooks {
-		// Check if hook was already executed
-		if containsJobForHook(jobs, hook) == true {
-			r.Log.V(4).Info("Skipping creation of job for hook '%s' as it already exists", hook.Name)
-			// Job was already created
-			continue
+		if len(jobs.Items) > 0 {
+			break // Job already exists
 		}
 
-		rawFileURL, err := r.PresignedGetURL(scan.UID, scan.Status.RawResultFile, defaultPresignDuration)
+		var rawFileURL string
+		rawFileURL, err = r.PresignedGetURL(scan.UID, scan.Status.RawResultFile, defaultPresignDuration)
 		if err != nil {
-			return err
+			break
 		}
-		findingsFileURL, err := r.PresignedGetURL(scan.UID, "findings.json", defaultPresignDuration)
+		var findingsFileURL string
+		findingsFileURL, err = r.PresignedGetURL(scan.UID, "findings.json", defaultPresignDuration)
 		if err != nil {
-			return err
+			break
 		}
 
-		jobName, err := r.createJobForHook(
-			&hook,
-			scan,
-			[]string{
-				rawFileURL,
-				findingsFileURL,
-			},
-		)
+		var jobName string
+		switch hook.Spec.Type {
+		case executionv1.ReadOnly:
+			jobName, err = r.createJobForHook(
+				&hook,
+				scan,
+				[]string{
+					rawFileURL,
+					findingsFileURL,
+				},
+			)
+		case executionv1.ReadAndWrite:
+			var rawFileUploadURL string
+			rawFileUploadURL, err = r.PresignedPutURL(scan.UID, scan.Status.RawResultFile, defaultPresignDuration)
+			if err != nil {
+				break
```

Comment:
The nested breaks are better than the previous duplication, but I propose that you always use switch labels whenever a switch is nested. Just to be very explicit and clear, and avoid defects.